pkg/s3select/select.go

```diff
@@ -220,25 +238,34 @@ func (reader *Input) processSelectReq(reqColNames []string, alias string, whereC
 		if condition {
 			// if its an asterix we just print everything in the row
 			if reqColNames[0] == "*" && functionNames[0] == "" {
-				rowStruct := &Row{
-					record: reader.printAsterix(record) + "\n",
+				switch inputType.(type) {
```

Comment:
This can be written as 

```
var row *Row

switch inputType.(type) {
case *CSVInput:
    row = &Row{}
case *JSONInput:
  row = &Row{}
}
myRow <- row
```

Reply:
Done